### ISSUE TITLE
Add example data JSON file(s) to upload using ORMs for staging data.

### ISSUE THREAD

**Issue by JacksonMaxfield** [CouncilDataProject/cdp-backend#8]
## Use Case
*Please provide a use case to help us understand your request in context*

Add a script for taking JSON file(s) and uploading them to a database using the upcoming ORMs (See #2).
Specifically this will be used to upload example data to a staging database and bucket.

## Solution
*Please describe your ideal solution*

We potentially need to have several sets of examples datasets.

* **Entirely minimal dataset**, looking through the database models, only upload data that is required
* **Minimal + random sample of optional values**
* **All database model fields filled**

Because of multiple levels of dataset, it may be best to have a single "all possible data" and then programmatically create the other levels of the dataset by just removing keys from the dataset when they are not required.

There should additionally be tests that the example data stored in the repo is always matching the current models.

## Alternatives
*Please describe any alternatives you've considered, even if you've dismissed them*

There is a potential argument to be made that this dataset is entirely programmatically created with some randomness to reduce data stored in git. But I think that just gets almost close to just creating a JSON file so seems like extra work.

**Comment by JacksonMaxfield** [comment]
Adding to this, it would be great to have tests that compare the example data to the database model (in the case that the data isn't generated), so that we know the example data is always compatible.

**Comment by tohuynh** [comment]
How should the JSON file be organized?
One big file for the entire database or one file for each collection? Which make things easier for #13?

For reference fields, I can use IDs as values? Or make it an object that contains the ID and the collection name?

Also for collection name, I'm going to use the names in the generated DB diagram. Sounds good?

**Comment by JacksonMaxfield** [comment]
I would personally say each collection having it's own file makes sense to me?

I think for "reference fields", generate fake GUIDs would be best and use those as ids. Then it will match firebase standard.

Example Body JSON
```json
{
"abcdef-123456-...": {
        "name": "Example Committee"
    }
}
```

Example Event JSON
```json
{
"123456-abcdef-...": {
        "body_ref": "abcdef-123456-..."
    }
}
```

The models have a `.collection_name` attr I _think_ those are the same as in the UML diagram but best to double check.

**Comment by JacksonMaxfield** [comment]
See `collection_name` attr in [docs](https://councildataproject.github.io/cdp-backend/cdp_backend.database.html#cdp_backend.database.models.Body.collection_name)

**Comment by tohuynh** [comment]
Some preliminary work: https://gist.github.com/tohuynh/e39c0e943710cf9ac850a60cf1cc3be4

The plan is to use topological sort on the DAG to get back a list of models. Then step through the list of models and create example data for each model. This works because in the reversed topological ordering whenever the loop encounters a ReferenceField, the example data for that ReferencField's model would have been already created. So the loop can use the already created GUID as a value for the ReferenceField. In creating the example data for each model, I will use the manually created json files and maybe randomly pick an already created GUID for any ReferenceField.

This sounds alright?

**Comment by JacksonMaxfield** [comment]
This is so well thought out and neat. I think this sounds and looks great.

I gave it a try and my only concern is how will you handle any of the file references. Some files are internal to firestore and some are external to the gov resource.

My thought on that is if you give me a big list of files to upload prior to running this script I am not concerned at all.

Additionally, when giving this a try I realized I needed to be on `main` instead off of my currently open `infrastructure-module` branch. I am going to finish that up and merge it tonight. The only difference that affects this script is that I am getting rid of the `_BUILD_IGNORE_CLASSES` list in favor of a `DATABASE_MODELS` list. `from cdp_backend.database import DATABASE_MODELS` which is a list of the classes themselves not the string names.

I will finish that up and merge the branch tonight so I am not in the way, but again the question of internal and external file management remains / in-general "value generation".

**Comment by JacksonMaxfield** [comment]
Also noting #13, I added a comment after we talked in Slack about having a `cdp-example` repo and I think your script should just be written as an `EventScraper` almost, so every time we fake a pipeline run it generates 5 new events if that makes sense.

**Comment by tohuynh** [comment]
>I gave it a try and my only concern is how will you handle any of the file references. Some files are internal to firestore and some are external to the gov resource.

>My thought on that is if you give me a big list of files to upload prior to running this script I am not concerned at all.

In the manually created file.json there will be bunch of files with all the fields. These will probably get added to example data first and have GUID created for them. Then when another model needs to fill in a file_ref field, it can use one of those created GUID. I guess I can use `media_type` to pick which File GUID to use?

>I will finish that up and merge the branch tonight so I am not in the way,

No rush, the script I wrote was just me experimenting ,trying things out.

**Comment by JacksonMaxfield** [comment]
Right. I am talking about the example transcript files I need to upload, and links to example seattle channel or other video resources, etc.

And no worries. PR is all merged now anyway :)

**Comment by tohuynh** [comment]
> Also noting #13, I added a comment after we talked in Slack about having a cdp-example repo and I think your script should just be written as an EventScraper almost, so every time we fake a pipeline run it generates 5 new events if that makes sense.

That makes sense. Basically, from the manually created json files, generate 5 new events and make it have the same format as what an EventScraper would produce?

**Comment by JacksonMaxfield** [comment]
Yep, exactly!

**Comment by JacksonMaxfield** [comment]
Correction / Note:

Because this example data is meant to be fed into an `EventProcessPipeline`, we don't need fake transcripts or audio files, etc, just links to videos / links to closed captions.

**Comment by JacksonMaxfield** [comment]
Migrating this issue to `cookiecutter-cdp-deployment`
